Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -259,43 +272,45 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
     ///
     init(
         config: ChatClientConfig,
+        tokenProvider: _TokenProvider<ExtraData>,
         workerBuilders: [WorkerBuilder],
         eventWorkerBuilders: [EventWorkerBuilder],
-        environment: Environment
+        environment: Environment,
+        completion: ((Error?) -> Void)? = nil
     ) {
         self.config = config
+        self.tokenProvider = tokenProvider
         self.environment = environment
         self.workerBuilders = workerBuilders
         self.eventWorkerBuilders = eventWorkerBuilders
-        
-        createBackgroundWorkers()
+
+        currentUserId = fetchCurrentUserIdFromDatabase()
         createWebSocketClient()
+        createAPIClient()
+
+        environment
+            .clientUpdaterBuilder(self)
+            .reloadUserIfNeeded(completion: completion)
```

Comment:
It took me a while to understand...
```swift
let updater = environment.clientUpdaterBuilder(self)
updater.reloadUserIfNeeded(completion: completion)
```

Reply:
done in c45940aca0a817c617215c7950bdf8554aa6d932